src/testing/mod.rs

```diff
@@ -248,33 +248,40 @@ pub trait SystemUnderTest<'a>: Default + Send + Sync {
         &mut self,
         rng: &mut ChaChaRng,
         ledger: &Arc<Mutex<MockLedger<'a, Self::Ledger, Self::MockNetwork>>>,
-    ) -> Keystore<'a, Self::MockBackend, Self::Ledger, LoaderMetadata> {
+    ) -> (Keystore<'a, Self::MockBackend, Self::Ledger, ()>, TempDir) {
+        let temp_dir = TempDir::new("test_keystore").unwrap();
         let mut loader = TrivialKeystoreLoader {
-            dir: TempDir::new("test_keystore").unwrap().into_path(),
+            dir: temp_dir.path().to_path_buf(),
+            rng: rng.clone(),
```

Comment:
`clone` clones the internal RNG state without updating `rng`. It's better to use `ChaChaRng::from_rng(rng)`, so that if, say, this function is called twice in a row, you will get two different keystores.